PittAPI/course.py

```diff
@@ -16,175 +16,103 @@
+            .replace(' ', '').replace('#', '_number')
+        if key.find('/') != - 1:
+            key = key[:key.index('/')]
+        header.append(key)
+    return header
+
+
+def _extract_course_data(header, course):
+    """Constructs a dictionary from column header labels(subject, class number, etc.) and course data."""
+    data = {}
+    for item, value in zip(header, course.findAll('td')):
+        data[item] = value.text.strip().translate({'\r': '', '\n': '', '\t': ''}).replace('\xa0', '')
+        if not data[item]:
+            data[item] = 'Not Decided'
+    try:
+        # TODO(azharichenko): Look into why there is an empty column header
+        del data['']
+    finally:
+        return data
 
 
 def get_class_description(term, class_number):
-    """
-    :returns: a string that is the description for CLASS_NUMBER in term TERM
-
-    :param: class_number: String, class number
-    :param: term: String, term number
-    """
-
-    url = 'http://www.courses.as.pitt.edu/detail.asp?CLASSNUM={}&TERM={}'.format(class_number, term)
-    page = requests.get(url)
-    soup = BeautifulSoup(page.text, 'lxml', parse_only=strainer)
-    table = soup.findChildren('table')[0]
-    rows = table.findChildren('tr')
-
-    has_description = False
-    for row in rows:
-        cells = row.findChildren('td')
-        for cell in cells:
-            if has_description:
-                return cell.string.strip()
-            if len(cell.contents) > 0 and str(cell.contents[0]) == '<strong>Description</strong>':
-                has_description = True
-
-
-def _get_course_dict(details):
-    course_details = []
-
-    if len(details) == 6:
-        course_details.append(
-            {
-                'subject': details[0],
-                'catalog_number': details[1],
-                'term': details[2].replace('\r\n\t', ' '),
-                'title': details[3],
-                'instructor': details[4] if len(details[4]) > 0 else 'Not decided',
-                'credits': details[5]
-            }
-        )
-    else:
-        course_details.append(
-            {
-                'subject': 'Not available',
-                'catalog_number': details[0],
-                'term': details[1].replace('\r\n\t', ' '),
-                'title': details[2].replace('\r\n\t', ' '),
-                'instructor': details[3] if len(details[3]) > 0 else 'Not decided',
-                'credits': details[4]
-            }
-        )
-
-    return course_details
-
-
-def _retrieve_from_url(url):
-    page = requests.get(url)
-    soup = BeautifulSoup(page.text, 'lxml', parse_only=strainer)
-    courses = soup.findAll("tr", {"class": "odd"})
-    courses_even = soup.findAll("tr", {"class": "even"})
-    courses.extend(courses_even)
-    return courses
+    """Return a string that is the description for class in a term"""
+    page = requests.get(URL + 'detail.asp?TERM={}&CLASSNUM={}'.format(_validate_term(term), class_number))
```

Comment:
`TERM` and `CLASSNUM` should be part of a `payload` dict and passed into `requests.get(url, params=payload)`